src/Calculator/Views/GraphingCalculator/GraphingCalculator.xaml.cpp

```diff
@@ -105,79 +109,100 @@ void CalculatorApp::GraphingCalculator::OnShareClick(Platform::Object ^ sender,
 // data to be shared. We will request the current graph image from the grapher as a stream that will pass to the share request.
 void GraphingCalculator::OnDataRequested(DataTransferManager ^ sender, DataRequestedEventArgs ^ args)
 {
-    auto resourceLoader = Windows::ApplicationModel::Resources::ResourceLoader::GetForCurrentView();
+    auto resourceLoader = ResourceLoader::GetForCurrentView();
+
     try
     {
-        // Get our title from the localized resources
-        auto EmptyEquationString = resourceLoader->GetString(L"EmptyEquationString");
-
-        std::wstring rawHtml = L"<p><img src='graph.png'></p>";
+        std::wstring rawHtml;
+        std::wstring graphImageHtml = L"<p><img src='graph.png' width='600'></p>";
 
         auto equations = ViewModel->Equations;
-        rawHtml += L"<p><table cellpadding=\"10\">";
-        rawHtml += L"<col width=\"20\">";
-        rawHtml += L"<row height=\"20\">";
-        for (unsigned i = 0; i < equations->Size; i++)
-        {
-            auto expression = equations->GetAt(i)->Expression->Data();
-            auto color = equations->GetAt(i)->LineColor->Color;
+        bool hasEquations = false;
 
-            if (equations->GetAt(i)->Expression->Length() == 0)
+        if (equations->Size > 0)
+        {
+            rawHtml += L"<span style=\"color: rgb(68, 114, 196); font-family: calibri, sans-serif; font-style: bold; font-size : 13pt;\">";
+            rawHtml += resourceLoader->GetString(L"EquationsShareHeader")->Data();
+            rawHtml += L"</span>";
+            rawHtml += L"<table cellpadding=\"0\">";
+            for (unsigned i = 0; i < equations->Size; i++)
             {
-                expression = EmptyEquationString->Data();
+                auto expression = equations->GetAt(i)->Expression;
+                auto color = equations->GetAt(i)->LineColor->Color;
+
+                if (expression->IsEmpty())
+                {
+                    continue;
+                }
+
+                hasEquations = true;
+
+                expression = GraphingControl->ConvertToLinear(expression);
+
+                std::wstring equatinColorHtml = L"color:rgb(";
+                equatinColorHtml += color.R.ToString()->Data();
+                equatinColorHtml += L",";
+                equatinColorHtml += color.G.ToString()->Data();
+                equatinColorHtml += L",";
+                equatinColorHtml += color.B.ToString()->Data();
+                equatinColorHtml += L");";
+
+                rawHtml +=
+                    L"<tr><td><span style=\"line-height: 0; font-size: 24pt;" + equatinColorHtml + L"\">&#x25A0;</span></td><td><div style=\"margin: 4px 0px 0px 0px; \">";
+                rawHtml += expression->Data();
```

Comment:
we should probably HTML encode this expression in case it uses some reserved characters.

Reply:
This is a good call out, do you know of an easy way to do this in c++/cx?